client/src/components/SandwichList.js

```diff
@@ -14,18 +19,65 @@ const SandwichList = (props) => {
       console.log(error)
     }
   }
+
+  const addSandwich = async (newSandwich) => {
+    try {
+      const response = await fetch("/api/v1/sandwiches", {
+        method: "POST",
+        headers: new Headers ({
+          "Content-Type": "application/json"
+        }),
+        body: JSON.stringify(newSandwich)
+      })
+      if (!response.ok) {
+        if (response.status === 422) {
+          const errorBody = await response.json()
+          const newErrors = translateServerErrors(errorBody.errors)
+          return setErrors(newErrors)
+        } else {
+          const errorMessage = `${response.status} (${response.statusText})`
+          const error = new Error(errorMessage)
+          throw error
+        }
+    } else {
+      const responseBody = await response.json()
+      setSandwiches([
+        ...sandwiches,
+        responseBody.newSandwich
+      ])
+    }
+    } catch (error) {
+      console.error(`Error in fetch: ${error.message}`)
+    }
+  }
   
   useEffect(() => {
     getSandwiches()
   }, [])
 
+  const toggleFormOnClick = event => {
+    setShowForm(true)
+  }
+
+  const formQuestion = 
+    (
+        <>
+          <ErrorList errors={errors}/>
+          <SandwichForm addSandwich={addSandwich} setErrors={setErrors}/>
+        </>
+    )
```

Comment:
this works! though you can actually put lines 64 - 67 on line 79 if you wish!

Reply:
We kind of liked it as is, so can we merge it like this? If not, we can change it.